Snake/Snake/Reader.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading;
+
+namespace Snake
+{
+    // This class is used as a key listner. It waits for a few milliseconds for a key to be pressed.
+    // Advantages of using this class: No busy waiting.
+
+    // Modified version of: http://stackoverflow.com/questions/57615/how-to-add-a-timeout-to-console-readline
+    class Reader
+    {
+        private static AutoResetEvent getInput, gotInput;

```

Comment:
does it need to be static? in general static is a bad smell. it leads to thread safety problems so avoid if you can.

also one variable per line. improves readability